### PR TITLE
Leader election only affects the controller not the metrics

### PR DESCRIPTION
**Pull request by mrIncompetent**

### CODE REVIEW

File: api/cmd/kubermatic-cluster-controller/main.go [hunk 1]
```diff
@@ -27,15 +28,16 @@ import (
...
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	kubeleaderelection "k8s.io/client-go/tools/leaderelection"
 	"k8s.io/client-go/tools/record"
 )
 
 var (
-	prometheusAddr        = flag.String("prometheus-address", "127.0.0.1:8085", "The Address on which the prometheus handler should be exposed")
+	prometheusAddr        = flag.String("prometheus-address", "127.0.0.1:8086", "The Address on which the prometheus handler should be exposed")
```
Comment (metalmatze) [inline api/cmd/kubermatic-cluster-controller/main.go]: Is there a reason this port changed? Either this changes and we update the helm chart or you revert to 8085.

File: api/cmd/kubermatic-cluster-controller/main.go [hunk 2]
```diff
@@ -131,8 +127,13 @@ func main() {
 
 		g.Add(func() error {
 			glog.Infof("Starting the internal http server: %s\n", *prometheusAddr)
-			return s.ListenAndServe()
+			err := s.ListenAndServe()
+			if err != nil {
+				return fmt.Errorf("internal http server failed: %v", err)
```
Comment (metalmatze) [inline api/cmd/kubermatic-cluster-controller/main.go:132]: Are you doing this just to give a little bit extra context?

Comment (metalmatze) [reply api/cmd/kubermatic-cluster-controller/main.go:132]: Which is fine, just asking.

**Review by metalmatze — CHANGES_REQUESTED** [review]

File: api/cmd/kubermatic-cluster-controller/main.go (hunk 1 above)
Comment (mrIncompetent) [reply api/cmd/kubermatic-cluster-controller/main.go]: Oh sit, sorry, i changed it to test a second instance :man_facepalming:

File: api/cmd/kubermatic-cluster-controller/main.go (hunk 2 above)
Comment (mrIncompetent) [reply api/cmd/kubermatic-cluster-controller/main.go:132]: Yes, i added a log message to every `interrupt` func

**Review by metalmatze — APPROVED** [review]

**Pull request merged**